pkg/apis/openebs.io/v1alpha1/upgrade_details.go

```diff
@@ -16,17 +16,43 @@ limitations under the License.
 
 package v1alpha1
 
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 // VersionDetails provides the details for upgrade
 type VersionDetails struct {
 	// If AutoUpgrade is set to true then the resource is
 	// upgraded automatically without any manual steps
 	AutoUpgrade bool `json:"autoUpgrade"`
-	// Current is the version of resource
-	Current string `json:"current"`
 	// Desired is the version that we want to
 	// upgrade or the control plane version
 	Desired string `json:"desired"`
+	// Status gives the status of reconciliation triggered
+	// when the desired and current version are not same
+	Status VersionStatus `json:"status"`
+}
+
+// VersionStatus is the status of the reconciliation of versions
+type VersionStatus struct {
 	// DependentsUpgraded gives the details whether all children
 	// of a resource are upgraded to desired version or not
 	DependentsUpgraded bool `json:"dependentsUpgraded"`
+	// Current is the version of resource
+	Current string `json:"current"`
+	// State is the state of reconciliation
+	State VersionState `json:"state"`
+	// Message is a human readable message if some error occurs
+	Message string `json:"message,omitempty"`
+	// Reason is the actual reason for the error state
+	Reason string `json:"reason,omitempty"`
+	// LastUpdateTime is the time the status was last  updated
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty"`
 }
+
+// VersionState is the state of reconciliation
+type VersionState string
+
+const (
```

Comment:
instead of 'ERROR', we can have like
'ReconcilePending',
'ReconcileInProgress',
'ReconcileDone' or 'Reconciled'
Lets keep these values in sync with K8S other status.Phase(s)..

Reply:
Can we take this up when we will update the state.

Reply:
Done